### PR TITLE
feat!: generate scores using static thresholding 

### PR DESCRIPTION
**Pull request by ab93**
Signed-off-by: Avik Basu <ab93@users.noreply.github.com>

- use payload factory with headers to instantiate the correct payload
- use numalogic v0.3.3
- generate scores using static threshold if no model is found
- changed the payload forwarding strategy - we now forward StreamPayload until threshold vtx with respective headers (MODEL_INFERENCE/STATIC_INFERENCE/MODEL_STALE) and in threshold vtx we conditional forward to postprocess and/or trainer vtx based on header

### CODE REVIEW

File: numaprom/udf/inference.py [hunk 1]
```diff
@@ -77,44 +81,49 @@ def _is_model_stale(
...
+        msgs = []
         train_payload = TrainerPayload(
-            uuid=stream_payload.uuid, composite_keys=OrderedDict(stream_payload.composite_keys)
+            uuid=payload.uuid, composite_keys=OrderedDict(payload.composite_keys)
         )
-        return [orjson.dumps(train_payload)]
+        msgs.append(orjson.dumps(train_payload))
+
+        # Calculate scores using static threshold
+        msgs.append(calculate_static_thresh(payload, STATIC_LIMIT))
```
Comment (nkoppisetty) [inline numaprom/udf/inference.py:115]: why are we doing this again here? if STATIC_INFERENCE we are just need to forward to next vertex right? which is done in line 95.

Comment (nkoppisetty) [reply numaprom/udf/inference.py:115]: We should just do this in threshold vertex, add a new field to `TrainPayload` like `stale_model` may be. If not stale model, that means no model at all, then calculate static threshold and send to postprocess.

File: numaprom/__init__.py [hunk 2]
```diff
@@ -17,4 +17,6 @@
 
 
 LOGGER.addHandler(stream_handler)
-logging.getLogger("pytorch_lightning").setLevel(logging.WARNING)
+pl_logger = logging.getLogger("pytorch_lightning")
+pl_logger.propagate = False
+pl_logger.setLevel(logging.ERROR)
```
[the remark this answers is not in the document]
Comment (ab93) [reply numaprom/__init__.py:22]: Sure, do you know where we get it? Is it in trainer?

File: numaprom/udf/threshold.py [hunk 3]
```diff
@@ -1,73 +1,77 @@
...
-    if train_payload:
-        _LOGGER.debug("%s - Previous clf not found. Sending to trainer..", train_payload.uuid)
-        return [(TRAIN_VTX_KEY, orjson.dumps(train_payload))]
+    # Load config
+    metric_config = get_metric_config(payload.composite_keys["name"])
 
-    recon_err = stream_payload.get_streamarray()
+    # Check if payload needs static inference
+    if payload.header == Header.STATIC_INFERENCE:
+        _LOGGER.debug("%s - Relaying forward static threshold payload")
```
Comment (ab93) [inline numaprom/udf/threshold.py:36]: Maybe a different log message here?

**Review by ab93 — COMMENTED** [review]
Approved from my side. Left one comment. Also I don't think I can approve my own PR.

**Review by vigith — APPROVED** [review]

**Pull request merged**